WPMedia/fusion-news-theme-blocks
# TMEDIA-212 - Update Overline to output sponsored content if needed

## Description
* Update overline shared style component to have sponsored content logic
* Update Overline block to use the shared-styles Overline component

## Jira Ticket
- [TMEDIA-212](https://arcpublishing.atlassian.net/browse/TMEDIA-212)

## Acceptance Criteria
_Making an Article Sponsored in Composer (`owner.sponsored:true`) replaced the Overline with the string "Sponsored Content", without adding a label. Adding a `label.basic` replaced the "Sponsored Content" string with a custom one, that is not linked._

## Test Steps

Content 
* Not sponsored - /local/2020/01/14/these-fish-tacos-channel-the-best-beach-vacation-vibes/
* Sponsored - /news/2020/04/15/7-questions-about-traveling-to-australia-during-catastrophic-fires-answered/

1. Checkout this branch `git checkout TMEDIA-212-overline-sponsored-content && git remote update --prune origin && git reset --hard origin/canary TMEDIA-212-overline-sponsored-content`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/extra-large-promo-block,@wpmedia/large-promo-block,@wpmedia/top-table-list-block,@wpmedia/overline-block,@wpmedia/shared-styles`
3. Visit article page and verify Overline shows "Sponsored Content" - http://localhost/news/2020/04/15/7-questions-about-traveling-to-australia-during-catastrophic-fires-answered/?_website=the-gazette
4. Visit article page and verify Overline shows section if content is not sponsored - http://localhost/local/2020/01/14/these-fish-tacos-channel-the-best-beach-vacation-vibes/?_website=the-gazette
5. Visit Homepage and verify items with sponsored content are shown on Promo blocks correctly - http://localhost/homepage/?_website=the-gazette
6. Using the content items listed above add a large promo and extra large promo block to a page and verify Overline shows correct output
7. For top table list test with "story-feed-query" and query `"7"`
8. For card list test with "story-feed-query" and query `"7 questions"`

## Effect Of Changes

Before = left, after = right 

<img width="1446" alt="TMEDIA-212-article-view" src="https://user-images.githubusercontent.com/868127/118806472-08177d00-b89f-11eb-947b-395949b949f3.png">

<img width="1877" alt="TMEDIA-212-promo-output" src="https://user-images.githubusercontent.com/868127/118806474-0948aa00-b89f-11eb-854a-d5166f65e6e2.png">


## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
